server/conn.go

```diff
@@ -110,9 +110,9 @@ func (cc *clientConn) handshake() error {
 func (cc *clientConn) Close() error {
 	cc.server.rwlock.Lock()
 	delete(cc.server.clients, cc.connectionID)
+	connGauge.Set(float64(len(cc.server.clients)))
```

Comment:
assign length to a variable and set metrics outside the lock will improve performance.